remotespark/livyclientfactory.py

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2015  aggftw@gmail.com
+# Distributed under the terms of the Modified BSD License.
+
+from base64 import b64encode
+
+from log import Log
+from connectionstringutil import get_connection_string_elements
+from livysession import LivySession
+from livyclient import LivyClient
+from reliablehttpclient import ReliableHttpClient
+
+
+class LivyClientFactory(object):
```

Comment:
Why do we need a separate factory class? I think we can move all of this logic to the construction of the LivyClient itself.